src/main/java/juuxel/libninepatch/NinePatch.java

```diff
@@ -186,55 +134,60 @@ static float lerp(float delta, float a, float b) {
     /**
      * Creates a {@link Builder} using a full texture.
      *
-     * @param texture the texture
-     * @param <T>     the texture type
+     * @param <T> the texture type
      * @return a nine-patch renderer builder using the texture
      */
-    public static <T> Builder<T> builder(T texture) {
-        return builder(new TextureRegion<>(texture, 0, 0, 1, 1));
-    }
-
-    /**
-     * Creates a {@link Builder} using a texture region.
-     *
-     * @param texture the texture region
-     * @param <T>     the texture type
-     * @return a nine-patch renderer builder using the texture region
-     * @throws NullPointerException if the texture region is null
-     */
-    public static <T> Builder<T> builder(TextureRegion<T> texture) {
-        return new Builder<>(texture);
+    public static <T> Builder<T> builder(Class<T> textureClass,
+                                         T topLeft, T topCenter, T topRight,
+                                         T centerLeft, T centerCenter, T centerRight,
+                                         T bottomLeft, T bottomCenter, T bottomRight) {
+        return new Builder<>(textureClass, topLeft, topCenter, topRight, centerLeft, centerCenter, centerRight, bottomLeft, bottomCenter, bottomRight);
     }
 
     /**
      * The rendering mode of a nine-patch renderer.
      */
     public enum Mode {
-        /** Regions of the texture are tiled to fill areas. The default value. */
+        /**
+         * Regions of the texture are tiled to fill areas. The default value.
+         */
         TILING,
-        /** Regions of the texture are stretched to fill areas. */
+        /**
+         * Regions of the texture are stretched to fill areas.
+         */
         STRETCHING
     }
 
     /**
      * A builder for {@link NinePatch}.
      *
      * @param <T> the texture type
-     * @see NinePatch#builder(Object)
-     * @see NinePatch#builder(TextureRegion)
+     * @see NinePatch#builder(Class, Object, Object, Object, Object, Object, Object, Object, Object, Object)
      */
     public static final class Builder<T> {
-        private final TextureRegion<T> texture;
+        private final T[][] textures;
         private int cornerWidth = 0;
         private int cornerHeight = 0;
-        private float cornerU = 0f;
-        private float cornerV = 0f;
```

Comment:
This functionality should remain as a fallback option for when texture regions for the sides and corners are not specified manually.